### ISSUE TITLE
Did you provide a single image inference script

### ISSUE THREAD

**Issue by CoinCheung** [mahyarnajibi/SNIPER#70]
I am looking through the code for a script or function that allows me to infer with one single image each time. Did you provide such function already in the code. If not, do you have any plan to add this ?

**Comment by bharatsingh430** [comment]
you can try the demo script and change the image

**Comment by CoinCheung** [comment]
Thanks, I should have been more careful. BTW, I notice there is a config field of ```CHIPS_DB_PARTS```. Will I suffer from accuracy loss if I change this to a smaller number, or it will simple slow down the training process while keeping the accuracy. I have an out of memory problem on my two 1080 gpus with ```USE_NEG_CHIPS``` set ```true```,  what is the best way to handle this problem ?

**Comment by bharatsingh430** [comment]
changing CHIPS_DB_PARTS will not change accuracy. Its just for reducing the memory consumption. If you increase it, CPU memory will reduce. To reduce GPU memory, I think you need to reduce the batch size.

**Comment by mahyarnajibi** [comment]
@CoinCheung ```BATCH_IMAGES``` is actually the number of images per GPU. The comment above it was wrong and is now corrected. So you need to reduce it to a lower number if you are getting a cuda out of memory error.

**Comment by CoinCheung** [comment]
Yes, I did not get that out of memory problem any more. However, I just wonder if your published results are trained with ```batch_size = 16``` per gpu, and will I suffer from an accuracy destroy with merely ```batch_size = 4```

**Comment by bharatsingh430** [comment]
4 may be a bit low, around 10 it should be fine

**Comment by CoinCheung** [comment]
@bharatsingh430  10 each gpu or 10 in total ?

**Comment by CoinCheung** [comment]
For me with weaker gpus, shall I turn off BN for the backbone network ?

**Comment by bharatsingh430** [comment]
10 on each gpu. I don't think BN is dependent on the GPU you have and it doesn't affect gpu memeory. It should be fine to train on 2 as well, you just need to adjust the learning rate.

**Comment by CoinCheung** [comment]
To confirm if I understand you correctly, the default configurations uses 8 gpus with each gpu 16 images. For me, I have two gpus and on each I assign 4 images. Thus by the linear rule, I should reduce my lr by 16, isn't it?

As for BN, the paper says one of the performance boosting tricks is to use BN. The default configuration has batch size of ```8 * 16 = 128``` while my configuration can merely afford ``` 2 * 4 = 8```. Besides, mxnet seems not able to support BN operation across gpus for what I know. So shall I turn it off to avoid the side effect brought by BN with a small batch size ? If so, is it enough that I only change from ```fix_bn=False``` to ```fix_bn=True``` in [this](https://github.com/mahyarnajibi/SNIPER/blob/master/symbols/faster/resnet_mx_101_e2e.py#L21) line with ```--cfg configs/faster/sniper_res101_e2e.yml``` option during training ?
